### PR TITLE
add alert for conntrack table getting full

### PR DESCRIPTION
**Pull request by sagor999**
## Related Issue(s)

Fixes this issue: https://github.com/gitpod-io/ops/issues/928

## Release Notes

```release-note
none
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **sagor999** after the PR has been reviewed.

Associated issue: *#928*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[operations/observability/mixins/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: operations/observability/mixins/cross-teams/rules/components/nodes/alerts.libsonnet
Comment (kylos101) [inline operations/observability/mixins/cross-teams/rules/components/nodes/alerts.libsonnet:39]: Rock on, @sagor999 !

May I ask you to [add a related runbook for this critical alert](https://github.com/gitpod-io/runbooks)? 

It's Gitpod policy to have a runbook for critical alerts, because they page. Warnings do not page (through PagerDuty) and do not require a runbook.

**Review by kylos101 — COMMENTED** [review]
Hey @sagor999 may I ask a favor?

Can you add a short Loom video showing how you tested that this warning and alert get tripped? This assumes the effort to setup such a condition is not crazy high. I ask because I'm not familiar with the conntrack table, or conditions that can cause it to get full.

Note: this assumes it is possible to manifest this condition in `core-dev`, but is not destructive. If it's neither possible, or possible but destructive, lemme know?

File: operations/observability/mixins/cross-teams/rules/components/nodes/alerts.libsonnet
Comment (sagor999) [reply operations/observability/mixins/cross-teams/rules/components/nodes/alerts.libsonnet:39]: Forgot to link it here, here is runbook for it: https://github.com/gitpod-io/runbooks/pull/320

**Comment by sagor999** [comment]
@kylos101 testing it in core-dev would be destructive. But what you can do is run this query in grafana explorer instead: 
```
(node_nf_conntrack_entries / on (pod) node_nf_conntrack_entries_limit / on (pod) group_right kube_pod_info) > 0.5
```
that will show you all nodes that used at least 50% of their conntrack table capacity.

File: operations/observability/mixins/cross-teams/rules/components/nodes/alerts.libsonnet
Comment (ArthurSens) [inline operations/observability/mixins/cross-teams/rules/components/nodes/alerts.libsonnet:61]:
```suggestion
               (node_nf_conntrack_entries / node_nf_conntrack_entries_limit) > 0.80
```

Any reason to not use a simplified query?

A join can make the query quite heavy and the added label, pod, doesn't provide any info, right? 🙂 
For a node that is running out of conntrack entries, there is no need to know which exporter pod is running, I believe you only care about what node is running out of entries

Comment (sagor999) [reply operations/observability/mixins/cross-teams/rules/components/nodes/alerts.libsonnet]: Yes, you are 100% right. Copy-paste error while testing this in grafana. Fixed!

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: b4aecbe4aac6efccd87c1e1ad7dd2c75d1855b16</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7931#issuecomment-1030860077" title="LGTM">kylos101</a>*

Associated issue: *#928*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[operations/observability/mixins/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/OWNERS)~~ [kylos101]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by mads-hartmann** [comment]
@kylos101 We have switched to using GH Review this morning, so this the /approve and /lgtm commands no longer have any effect. You'll have to approve this in Github before it can be merged ☺️

**Review by kylos101 — APPROVED** [review]
LGTM

**Review by kylos101 — APPROVED** [review]
LGTM

**Pull request merged**
